server/api/rule.go

```diff
@@ -42,6 +43,42 @@ func newRulesHandler(svr *server.Server, rd *render.Render) *ruleHandler {
 	}
 }
 
+type ruleMiddleware struct {
+	s  *server.Server
+	rd *render.Render
+	*server.Handler
+}
+
+func newRuleMiddleware(s *server.Server, rd *render.Render) ruleMiddleware {
+	return ruleMiddleware{
+		s:       s,
+		rd:      rd,
+		Handler: s.GetHandler(),
+	}
+}
+
+func (m ruleMiddleware) Middleware(h http.Handler) http.Handler {
```

Comment:
Will we support it in scheduling server?

Reply:
scheduling server uses `gin` and does implement this in `Handler`. So i think we don't need to use middleware in scheduling server now.